src/type-translator.ts

```diff
@@ -223,26 +224,41 @@ export class TypeTranslator {
       case ts.TypeFlags.Object:
         return this.translateObject(type as ts.ObjectType);
       case ts.TypeFlags.Union:
-        let unionType = type as ts.UnionType;
-        let parts = unionType.types.map(t => this.translate(t));
-        // Union types that include boolean literals and other literals can
-        // end up repeating the same Closure type. For example: true | boolean
-        // will be translated to boolean | boolean. Remove duplicates to produce
-        // types that read better.
-        parts = parts.filter((el, idx) => parts.indexOf(el) === idx);
-        return parts.length === 1 ? parts[0] : `(${parts.join('|')})`;
+        return this.translateUnion(type as ts.UnionType);
       case ts.TypeFlags.Intersection:
       case ts.TypeFlags.Index:
       case ts.TypeFlags.IndexedAccess:
         // TODO(ts2.1): handle these special types.
         this.warn(`unhandled type flags: ${ts.TypeFlags[type.flags]}`);
         return '?';
       default:
+        // Handle cases where multiple flags are set.
+
+        // Booleans are represented as
+        //   ts.TypeFlags.Union | ts.TypeFlags.Boolean
+        // where the union is a union of true|false.
+        // Note also that in a more complex union, e.g. boolean|number, then
+        // it's a union of three things (true|false|number) and
+        // ts.TypeFlags.Boolean doesn't show up at all.
+        if (type.flags & ts.TypeFlags.Union) {
```

Comment:
I am confused here. You say that for booleans Union and Boolean flags are on. Since more than one flag is one we are in the 'default' case. Then we are in this case too, and we output 'true | false', but don't we want 'boolean' for booleans?